README.md

```diff
@@ -24,95 +22,135 @@ Well, it's our lucky day!
-  debugger
-  console.log(`${even} is not odd!`)
-})
+  debugger;
+  console.log(`${even} is not odd!`);
+});
 ```
 
-(As a reminder, `debugger` is globally available and opens up the debug console in your browser when it's evaluated. Clicking the arrow will let you jump around and inspect the arguments to `forEach`'s function argument on as it iterates over the `evens` array.)
+As a reminder, `debugger` is globally available and opens up the debug console in your browser when it's evaluated. Clicking the arrow (<img width="75" alt="step-through arrow" src="https://user-images.githubusercontent.com/17556281/27013165-b7852f78-4eac-11e7-80f7-2a97a40633b3.png">) or pressing the `F10` key will let you step through the function calls and inspect the three arguments that `.forEach()` passes to the function as it iterates over the `evens` array. On each pass through the `.forEach()` loop, the `even`, `index`, and `array` variables contain the following values:
 
-This is a _super_ powerful idea that makes it much easier for us to build programs that can do _many_ things. Instead of having to operate on each element using a `for` loop, we can instead very succinctly pass a function to `Array.prototype.forEach` and let the function do the work for us!
+| Pass # | `even` | `index` |       `array`       |
+|:------:|:------:|:-------:|:-------------------:|
+|   1    |   0    |    0    | [0, 2, 4, 6, 8, 10] |
+|   2    |   2    |    1    | [0, 2, 4, 6, 8, 10] |
+|   3    |   4    |    2    | [0, 2, 4, 6, 8, 10] |
+|   4    |   6    |    3    | [0, 2, 4, 6, 8, 10] |
+|   5    |   8    |    4    | [0, 2, 4, 6, 8, 10] |
+|   6    |   10   |    5    | [0, 2, 4, 6, 8, 10] |
 
-## Functions as Arguments
+It's possible to provide the same functionality with a regular `for` loop, but to do so we would have to jump through some hoops that `.forEach()` mercifully abstracts away for us. To replicate `.forEach()` with a `for` loop, we could define the function ahead of time and then invoke it inside of the loop...
 
-When a language allows functions to be used as arguments, we say that the language treats functions as **first-class objects**. Oftentimes, these functions serve as **callbacks**, which do just what their name implies: they are _called back_, potentially with arguments, and do their work on bits of data that weren't available outside of the function.
+```js
+var evens = [0, 2, 4, 6, 8, 10];
+
+function evenFn(even, index, array) {
+  console.log(`${even} is not odd!`);
+}
+
+for (let i = 0; i < evens.length; i++) {
+  evenFn(evens[i], i, evens);
+}
+```
 
-For example, let's say that we have a function for logging the square of a number called, appropriately enough, `square`.
+...but that seems like too many extra lines for such a simple operation. We could also use an IIFE, an immediately-invoked function expression, to make our code a bit more compact, but it's starting to get far more confusing than the elegant simplicity of a `.forEach()`:
```

Comment:
Drop the bit about IIFE. They are going to do research and just generally go down the rabbit hole that is IIFEs. Either we have to do a whole thing on it..or not explain it. Remember this is early curriculum for some